cloudevents/extensions/verifiability.md

```diff
@@ -0,0 +1,290 @@
+in transit.
+
+This proposal only applies to individual events. It does not give consumers any
+guarantees about the completeness of the event stream or the order of events.
+
+The threats of malicious actors removing or hiding items from the event stream
+as well as swapping their order are not addressed by this proposal. Neither are
+the possibilities of messages accidentally getting lost or delivered in the
+wrong order. Both can be addressed by producers through means of adding the
+necessary information inside the event payloads.
+
+Further, this proposal only aims at *verifiability*. It does not aim to enable
+*confidentiality*. Consequently, it does not address the threat of unauthorized
+parties reading CloudEvents that were not meant for them.
+
+## Notational Conventions
+
+As with the main [CloudEvents specification](../spec.md), the key words "MUST",
+"MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT",
+"RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as
+described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+However, the scope of these key words is limited to when this extension is
+used. For example, an attribute being marked as "REQUIRED" does not mean
+it needs to be in all CloudEvents, rather it needs to be included only when
+this extension is being used.
+
+## Attributes
+
+### verificationmaterial (Verification Material)
+    - Type: `String`
+    - Description: the material that consumers can use to verify a CloudEvent.
+      For example, the verification material could be a signature of the event
+      based on a private key. Consumers would use the producers’ public key to
+      verify the signature.
+    - Constraints:
+        - OPTIONAL
+        - If present, MUST be base64 encoded
+        - If present, a `verificationmaterialtype` attribute MUST also be
+          present
+
+### verificationmaterialtype (Verfication Material Type)
+    - Type: `String`
+    - Description: tells consumers of CloudEvents what type the verification
+      material is of. This high level category is used by the consumer to
+      choose an appropriate implementation for performing the verification. The
+      verification material can contain information specific to the
+      implementation such as a sub type, version information, etc.
+    - Constraints:
+        - OPTIONAL
+        - If present, a `verificationmaterial` attribute MUST also be present
+        - Producer and consumer MUST agree on an appropriate type
+
+## Usage
+
+When this extension is used, producers MUST set both the `verificationmaterial`
+and `verificationmaterialtype` attributes. Consumers can choose to verify if
+the material.
+
+## Assumptions
+
+This proposal contains a few assumptions that will be highlighted here.
+
+1. SDKs will verify as early as possible which depends on the
+   verification implementation.
+2. Users manage their secrets, e.g. public key infrastructure (PKI).
```

Comment:
Can you elaborate on why these assumptions are being made? It'll help people understand the overall proposal.

Reply:
In particular, when option 1 is chosen, does it impact assumption 1 ?